### PR TITLE
Add configurable historical market data behavior and updated caching.

### PR DESCRIPTION
**Pull request by stevendavis**

### CODE REVIEW

**Comment by eprbell** [comment]
Cool! Will review this in the weekend.

**Comment by eprbell** [comment]
I didn't have a chance to do a detailed review yet, but I have some early feedback. I don't think we should have weighted historical prices, because they are synthetic and not reflected in any record. My concern is that artificial numbers will not be accepted during an IRS audit, so we should avoid them. On the other hand HLOC numbers that are produced by the endpoint can be defended easily (we should record them in the debug log, BTW). The concept of "nearest" is interesting and doesn't violate the above principle, so we can keep it.

**Review by eprbell — CHANGES_REQUESTED** [review]
I finished a first review iteration: a few changes needed, but looks pretty good. Once this is merged I can build the new pair conversion plugin infrastructure on top of this.

File: src/dali/dali_configuration.py
Comment (eprbell) [reply src/dali/dali_configuration.py:147]: Yeah, you're right: all keywords are collected in one enum at the top and this is a set of values instead (like `DIRECTION_SET` or `_FIAT_FIELD_SET`. 
I suggest a couple of things:
* use dict instead of Tuple (see StackOverflow link above) or alternatively use a plain Set and just sort it on the fly when you print to the logger (either option works for me, no strong opinion)
* rename it to HISTORICAL_PRICE_KEYWORD_SET, for consistency with other similar sets in the file

**Comment by eprbell** [comment]
Looking solid. I noticed the changes in the ods files: can you describe what needed to change there?

**Comment by stevendavis** [comment]
The change to the ods file was the note column.  For example:
```
-spot_price read from Coinbase Pro;
+high spot_price read from Coinbase Pro;
```

**Comment by eprbell** [comment]
> The change to the ods file was the note column. For example:
> 
> ```
> -spot_price read from Coinbase Pro;
> +high spot_price read from Coinbase Pro;
> ```

Makes sense. There is one [last remaining comment](https://github.com/eprbell/dali-rp2/pull/44#discussion_r864305446), then I think we're ready to merge.

**Comment by stevendavis** [comment]
What requested changes are still remaining?  I think all the comments have been resolved.  Did I miss something?

**Comment by eprbell** [comment]
> What requested changes are still remaining? I think all the comments have been resolved. Did I miss something?

No, it's all good: I just want to do one final review and then we're good to go.

**Comment by eprbell** [comment]
LGTM, thanks for submitting this PR!

**Pull request merged**
